include/SolverAlgorithm.h

```diff
@@ -65,7 +66,8 @@ struct NGPApplyCoeff
   mutable stk::mesh::NgpField<double> diagField_;
   stk::mesh::NgpField<int> iblankField_;
 
-  CoeffApplier* deviceSumInto_;
+  Teuchos::RCP<CoeffApplier> deviceSumInto_;
```

Comment:
This concerns me. I think we need a generic CoeffApplier definition.

Reply:
This has been reverted. It does look like this was the main culprit for the double-free issue we were encountering.